packages/backend-storage/src/factory.ts

```diff
@@ -39,22 +29,18 @@ class AmplifyStorageFactory
   /**
    * Get a singleton instance of the Bucket
    */
-  getInstance = ({
-    constructContainer,
-    outputStorageStrategy,
-    importPathVerifier,
-  }: ConstructFactoryGetInstanceProps): AmplifyStorage => {
+  getInstance = (
+    getInstanceProps: ConstructFactoryGetInstanceProps
+  ): AmplifyStorage => {
+    const { constructContainer, importPathVerifier } = getInstanceProps;
     importPathVerifier?.verify(
       this.importStack,
       path.join('amplify', 'storage', 'resource'),
       'Amplify Storage must be defined in amplify/storage/resource.ts'
     );
     this.validateName(this.props.name);
     if (!this.generator) {
-      this.generator = new AmplifyStorageGenerator(
-        this.props,
-        outputStorageStrategy
-      );
+      this.generator = new StorageGenerator(this.props, getInstanceProps);
```

Comment:
It's weird to have `StorageGenerator` inside a `StorageFactory`. Maybe `StorageContainerEntryGenerator`?